drivers/dp/swdp_ll_pin_stm32.h

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2025 Google LLC
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <stm32_ll_gpio.h>
+
+#define CPU_CLOCK CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC
+#define FAST_BITBANG_HW_SUPPORT 1
+
+static ALWAYS_INLINE void swdp_ll_pin_input(void *const base, uint8_t pin)
+{
+	GPIO_TypeDef *gpio = base;
+
+	LL_GPIO_SetPinMode(gpio, BIT(pin), LL_GPIO_MODE_INPUT);
```

Comment:
I think these partial write accesses in GPIO regrsiters should be protected by a semaphore.
Could we use here the semaphore used in **gpio_stm32_.c**?
```c
	z_stm32_hsem_lock(CFG_HW_GPIO_SEMID, HSEM_LOCK_DEFAULT_RETRY);
	(...)
	z_stm32_hsem_unlock(CFG_HW_GPIO_SEMID);